src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -783,8 +783,23 @@ public int GetENCBuildState(ShellInterop.ENC_BUILD_STATE[] pENCBuildState)
                     }
                     else if (_encService.EditSession != null)
                     {
-                        _projectBeingEmitted = _vsProject.VisualStudioWorkspace.CurrentSolution.GetProject(_vsProject.Id);
-                        _lastEditSessionSummary = GetProjectAnalysisSummary(_projectBeingEmitted);
+                        // Fetch the latest snapshot of the project and get an analysis summary for any changes 
+                        // made since the break mode was entered.
+                        var currentProject = _vsProject.VisualStudioWorkspace.CurrentSolution.GetProject(_vsProject.Id);
```

Comment:
The EditAndContinueDiagnosticUpdateSource instance is called once from BuildForEnc, which we don't believe will be called when there are no changes.  The other call is when we're trying to report that we were unable to determine the MVID for the output of a project.  It seems like that might cause a crash, but I don't believe we've seen any (and it looks like it would happen before the one this PR works around).